### ISSUE TITLE
Add Entry Crashes

### ISSUE THREAD

**Issue by searchy2** [zeroturnaround/zt-zip#92]
I am trying to use AddEntry to add an SQLite database to a zip file. The code is as follows:
   
```
String dbFilePath = mContext.getDatabasePath("NotesDB").getPath();
File dbFile = new File(dbFilePath);
ZipUtil.addEntry(new File(saveFilePath), "NotesDB.db", dbFile, new File(saveFilePath));
```

It does not work and crashes with the following error:
```
java.lang.NoSuchMethodError: No virtual method getCreationTime()Ljava/nio/file/attribute/FileTime; in class Ljava/util/zip/ZipEntry; or its super classes (declaration of 'java.util.zip.ZipEntry' appears in /system/framework/core-oj.jar)
                                                                      at org.zeroturnaround.zip.timestamps.Java8TimestampStrategy.setTime(Java8TimestampStrategy.java:31)
                                                                      at org.zeroturnaround.zip.ZipEntryUtil.copyEntry(ZipEntryUtil.java:109)
                                                                      at org.zeroturnaround.zip.ZipEntryUtil.copyEntry(ZipEntryUtil.java:91)
                                                                      at org.zeroturnaround.zip.ZipUtil$10.process(ZipUtil.java:1964)
                                                                      at org.zeroturnaround.zip.ZipUtil.iterate(ZipUtil.java:409)
                                                                      at org.zeroturnaround.zip.ZipUtil.copyEntries(ZipUtil.java:1960)
                                                                      at org.zeroturnaround.zip.ZipUtil.addEntries(ZipUtil.java:1810)
                                                                      at org.zeroturnaround.zip.ZipUtil.addEntries(ZipUtil.java:1782)
                                                                      at org.zeroturnaround.zip.ZipUtil.addEntry(ZipUtil.java:1744)
                                                                      at org.zeroturnaround.zip.ZipUtil.addEntry(ZipUtil.java:1636)
```

I suspect I might be doing something wrong but I do not know what it is. Thanks for the great library!

**Comment by toomasr** [comment]
zt-zip supports more features on later Java versions. I see that zt-zip thinks it is running in Java8 mode but cannot find the method which could mean that actually the java8 detection has failed and since you are running an earlier version the method doesn't indeed exist as the exception states.

Can you double check which Java version you are running? Easiest is to put the following lines in the code and paste the output.

`System.out.println(System.getProperty("java.vendor"));
System.out.println(System.getProperty("java.version"));`

Also, which zt-zip version are you running?

**Comment by searchy2** [comment]
Thanks for the quick response and helping me troubleshoot the issue. 
I ran this in the emulator and received the following output:
java.vendor: The Android Project
java.version: 0

So the emulator's java version is not returning the correct value.

I am using v1.9 of zt-zip.

**Comment by toomasr** [comment]
Understood, I think we need to fix how we go about detecting the version then. The Android environment kind of surprises the detection.

@reinra you have been doing a fair bit of Android lately. Do you mind taking a stab at it? I can do it myself but testing will be somewhat difficult for me.

**Comment by searchy2** [comment]
Thanks for looking into this issue. As it is, this issue appears to only affect emulator devices so it is not that critical. That said, it would be nice to be able to test on emulators!

**Comment by reinra** [comment]
Should be fixed now.

**Comment by toomasr** [comment]
We've released a new version 1.12 that has the fix in.

**Comment by searchy2** [comment]
Wow! Nice work, really loving this library! It works amazing for creating an in-app backup function. Now my users won't lose their data anymore!
